pkg/stanza/fileconsumer/file_sort.go

```diff
@@ -0,0 +1,206 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package fileconsumer // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer"
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"time"
+
+	strptime "github.com/observiq/ctimefmt"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/timeutils"
+)
+
+const (
+	SortTypeNumeric      = "numeric"
+	SortTypeTimestamp    = "timestamp"
+	SortTypeAlphabetical = "alphabetical"
+)
+
+type SortRule struct {
+	Regex     string `mapstructure:"regex,omitempty"`
```

Comment:
I recall discussing this offline but didn't we decide it's better to have a single regex that all rules refer to? This would allow the future `group_by` to also use this. I think the idea would be that you name capture groups and then refer to one in each sort rule.